Wrappers/Python/cil/optimisation/operators/GradientOperator.py

```diff
@@ -311,8 +349,13 @@ def datacontainer_as_c_pointer(x):
     def ndarray_as_c_pointer(ndx):
         return ndx.ctypes.data_as(c_float_p)
         
-    def direct(self, x, out=None):    
-        ndx = np.asarray(x.as_array(), dtype=np.float32, order='C')
+    def direct(self, x, out=None): 
+        
+        # Case for psedo 3D data with only one slice
+        if self.is2D:            
+            ndx = np.squeeze(np.asarray(x.as_array(), dtype=np.float32, order='C'))
```

Comment:
I haven't understood if the `squeeze` is strictly necessary, however numpy's documentation says "This is always a itself or a view into a." so it won't allocate new memory.

I suppose that @ckolbitsch is going to say that this won't work with complex data! Probably up in the `GradientOperator` definition we should set backend numpy if complex data.

Reply:
If I do not squeeze the `cilacc.fdiff3D` will be called which is not correct.